orchagent/qosorch.cpp

```diff
@@ -564,6 +566,141 @@ type_map& QosOrch::getTypeMap()
     return m_qos_type_maps;
 }
 
+void QosOrch::initColorACL()
+{
+    SWSS_LOG_ENTER();
+    sai_object_id_t acl_table_id;
+
+    acl_table_id = initSystemACLTable();
+    if (acl_table_id != 0)
+    {
+        // dscp 8, ecn 0 --> yellow
+        initACLEntryForECN(acl_table_id, 1000, 0x00, 0x08, SAI_PACKET_COLOR_YELLOW);
+        // dscp 0, ecn 0 --> yellow
+        initACLEntryForECN(acl_table_id,  999, 0x00, 0x00, SAI_PACKET_COLOR_YELLOW);
+    }
+
+    return;
+}
+
+sai_object_id_t QosOrch::initSystemACLTable()
+{
+    SWSS_LOG_ENTER();
+    vector<sai_attribute_t> attrs;
+    sai_attribute_t attr;
+    sai_object_id_t acl_table_id;
+    sai_status_t status;
+
+    // create system acl table
+    attr.id = SAI_ACL_TABLE_ATTR_STAGE;
+    attr.value.s32 = SAI_ACL_STAGE_INGRESS;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_PRIORITY;
+    attr.value.u32 = 10;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_DST_MAC;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_SRC_IP;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_DST_IP;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_IP_PROTOCOL;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_L4_SRC_PORT;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_L4_DST_PORT;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_ECN;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_DSCP;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_ACL_TABLE_ATTR_FIELD_RANGE;
```

Comment:
Oct 21 22:38:10 xxx-xxxxx-xx-x ERR orchagent: :- meta_generic_validation_create: ACL_TABLE:SAI_ACL_TABLE_ATTR_FIELD_RANGE:INT32 attribute is mandatory but not passed in attr list
Oct 21 22:38:10 xxx-xxxxx-xx-x ERR orchagent: :- initSystemACLTable: create system acl table. sai_acl_api->create_acl_table failed: -14
Oct 21 22:38:10 xxx-xxxxx-xx-x INFO docker[2033]: terminate called after throwing an instance of 'std::runtime_error'